crates/nu-engine/src/nu_variable.rs

```diff
@@ -0,0 +1,219 @@
+use crate::scope::create_scope;
+use core::fmt;
+use nu_protocol::{
+    engine::{EngineState, Stack},
+    HistoryFileFormat, LazyRecord, ShellError, Span, Value,
+};
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+use sysinfo::SystemExt;
+
+// a LazyRecord for the special $nu variable
+// $nu used to be a plain old Record, but LazyRecord lets us load different fields/columns lazily. This is important for performance;
+// collecting all the information in $nu is expensive and unnecessary if  you just want a subset of the data
+#[derive(Serialize, Deserialize)]
+pub struct NuVariable {
+    #[serde(skip)]
+    pub engine_state: EngineState,
+    #[serde(skip)]
+    pub stack: Stack,
```

Comment:
Not sure do we need the ownership of `EngineState` and `Stack`?  What about just take reference of it?

Reply:
That gets difficult with lifetimes... might be nice to get that working eventually though

Reply:
You might check the ScopeData struct that stores reference to EngineState and Stack https://github.com/nushell/nushell/blob/main/crates/nu-engine/src/scope.rs . But yeah, it might be a pickle having the references stored inside Value without spilling the lifetime annotations everywhere.

An alternative: Don't store the EngineState and Stack at all and instead have `get_column_map(&self, engine_state: &EngineState, stack: &Stack)`. Could we do that?

Reply:
> An alternative: Don't store the EngineState and Stack at all and instead have `get_column_map(&self, engine_state: &EngineState, stack: &Stack)`. Could we do that?

I suspect that it will be difficult to thread the EngineState and Stack through everywhere cell paths are evaluated. But it's worth a try.